### PR TITLE
wgsl: Make builtin returned struct names consistent

### PR DESCRIPTION
**Pull request by ben-clayton**
Apply a `_f32` suffix to `f32` overloads - there's no reason for `f32` to be special.

Fix a couple of typos along the way.

### CODE REVIEW

File: wgsl/index.bs [hunk 1]
```diff
@@ -12805,7 +12805,7 @@ but a value may infer the type.
 
     Returns the `__modf_result_abstract` built-in structure, defined as follows:
     ```rust
-struct __modf_result {
+struct __modf_result_abstract {
```
Comment (mehmetoguzderin) [inline wgsl/index.bs:12808]: I think `_f32` suffix is better but could it make sense to omit `_abstract`?

File: wgsl/index.bs [hunk 2]
```diff
@@ -12912,7 +12912,7 @@ but a value may infer the type.
     The `i`'th component of the whole and fractional parts equal the whole and fractional parts
     of `modf(e[i])`.
 
-    Returns the `__modf_result_vecN` built-in structure, defined as follows:
```
Comment (dneto0) [inline wgsl/index.bs:12915]: thanks for this catch.

**Review by dneto0 — APPROVED** [review]

File: wgsl/index.bs (hunk 1 above)
Comment (mehmetoguzderin) [reply wgsl/index.bs:12808]: @dneto0 I kind of like how `__atomic_compare_exchange_result<T>` does it (where we could contraint T to be AbstractFloat), but it is fine either way, esp. if it is thought that no `
_abstract` adds ambiguity despite members.

Comment (mehmetoguzderin) [reply wgsl/index.bs:12808]: (oops didn't notice this got merged)

**Pull request merged**